### PR TITLE
"Get" Provider Testing

### PR DESCRIPTION
**Pull request by anthonylasso**
# Description
This PR adds test coverage to the various "get" methods in `register.py`

It increases the `register.py` coverage by 7%, and total repo coverage by 1%.

Is related to issue [enterprise-#62](https://github.com/featureform/featureform-enterprise/issues/62)

`Passing tests`
Screenshot 2023-08-11 at 2 26 05 PM

`register.py coverage increase`

**PR 59%**
Screenshot 2023-08-11 at 2 43 11 PM

**Main 52%**
Screenshot 2023-08-11 at 2 43 01 PM

`repo coverage increase`

**PR 56%**
Screenshot 2023-08-11 at 2 50 21 PM

**Main 55%**
Screenshot 2023-08-11 at 2 49 33 PM

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

**Review by sdreyer — APPROVED** [review]

File: client/src/featureform/register.py [hunk 1]
```diff
@@ -1582,31 +1580,29 @@ def get_source(self, name, variant, local=False):
                 path="",
             )
         else:
-            fakeDefinition = PrimaryData(location=SQLTable(name=""))
-            fakeSource = Source(
+            mock_definition = PrimaryData(location=SQLTable(name=""))
```
Comment (aolfat) [inline client/src/featureform/register.py:1583]: maybe add a comment why we need this in the first place, to see non test code have a mock def(n) is pretty confusing.

emptyDefinition seems more suitable to me

File: client/src/featureform/resources.py [hunk 2]
```diff
@@ -453,12 +453,12 @@ def serialize(self) -> bytes:
 @typechecked
 @dataclass
 class MongoDBConfig:
-    username: str
-    password: str
-    host: str
-    port: str
-    database: str
-    throughput: int
+    username: str = ""
```
Comment (aolfat) [inline client/src/featureform/resources.py:456]: do we really want users to be able to not pass username or password to any of these?

Comment (aolfat) [reply client/src/featureform/resources.py:456]: and all the other fields

File: client/src/featureform/register.py [hunk 3]
```diff
@@ -1722,13 +1712,13 @@ def get_postgres(self, name):
...
+        mock_config = PostgresConfig(
+            host="", port="", database="", user="", password=""
+        )
+        mock_provider = Provider(
+            name=name, function="OFFLINE", description="", team="", config=mock_config
         )
-        return OfflineSQLProvider(self, fakeProvider)
+        return OfflineSQLProvider(self, mock_provider)
 
     def get_snowflake(self, name):
```
Comment (aolfat) [inline client/src/featureform/register.py:1723]: I'm actually really confused what these functions do again.

File: client/src/featureform/register.py (hunk 1 above)
Comment (anthonylasso) [reply client/src/featureform/register.py:1583]: prep for future work. we're going to fill these guys in in the coming weeks.

File: client/src/featureform/register.py (hunk 3 above)
Comment (anthonylasso) [reply client/src/featureform/register.py:1723]: prep work for now, they just return resources, eventually we'd like them to return the data as well.

File: client/src/featureform/resources.py (hunk 2 above)
Comment (anthonylasso) [reply client/src/featureform/resources.py:456]: on second thought good point. these should be required fields. I'll edit and see if anything blows up.

**Pull request merged**